README.md

```diff
@@ -235,21 +235,11 @@ class MyModel(models.Model):
     username = models.CharField(max_length=100)
 
 
-def func(m: WithAnnotations[MyModel]) -> str:
```

Comment:
I don't think that we should remove this feature. Because I see that a lot of users discuss this and use this. Maybe we can treat `WithAnnotations[MyModel]` as `WithAnnotations[MyModel, Any]` implicitly? Via type var defaults or something?

This way we can continue to support this code (maybe with a deprecation warning, why not). But breaking it - does not seem right to me :(

Reply:
But, I agree that we should not have done this in the first place.

Reply:
Yeah, I suppose, gonna have to see if I can get it working. The support for any/arbitrary attributes is a bit of work I think

Reply:
We only need to support `Any`, no other arguments / types.

Reply:
Yes, the `Any` type argument is fine. But that doesn't add support for accessing an arbitrary attribute, like the previous implementation supported.

Essentially you'll still get something like:

```python
def func(m: WithAnnotations[MyModel]) -> str:
    return m.asdf  # E: MyModel@AnnotatedWith[Any] has no attribute "asdf"
```

I don't have any solution that removes the "has no attribute" error.

Reply:
Just to clarify: the current implementation replaces `WithAnnotations[MyModel]` with `MyModel`. So we support writing `WithAnnotations[MyModel]` but will now yield "has no attribute" errors on accessing attributes unknown to `MyModel`

Reply:
We can add `__getattr__(self, name: str) -> Any` to `Model@AnnotatedWith[Any]` types. This way all known attributes will have correct types, all unknown ones can be used. I will also suggest to add an options `--strict-model-annotations` with `False` as default for backward compatibility. What do you think?

Reply:
There's 1 problem that I see with that, unless I'm missing something, and `Model@AnnotatedWith[Any]` is an `Instance` and not a `TypeInfo`. But we need to add the `__getattr__` method to the `TypeInfo` though if we do, all instances of `Model@AnnotatedWith` gets the `__getattr__` method

Reply:
Won't `.extra_attrs` do the job for single instance `__getattr__`? 🤔 
I am not sure, just asking.

Reply:
`__getattr__` will work for the arbitrary attributes. But we still want/need to be able to match the model type on e.g. a function argument.

This whole thing is kind of twofold.

1. Subclassing the model and the generic TypedDict argument for proper matching for `WithAnnotations` as a function argument
2. `extra_attrs` for accessing declared annotated attributes e.g. inside a function body

`__getattr__` would help out with attributes accessed inside the body. But I can't find a way to add it and keep __1.__ working and not affect all other instances. I've also tried adding the `__getattr__` method to `extra_attrs` with no luck: `ExtraAttrs(attrs={"__getattr__": CallableType(...)}, ...)`

There's also the `TypeInfo.fallback_to_any` which is the same as subclassing `Any` or our `AnyAttrAllowed`. But same issue here, it affects the type and not instances of said type.

Reply:
Thanks a lot for your research! Not really what I hoped for :(

In this case, let's do the cleanest solution and break `WithAnnotations[M]` usage in a way that we restrict known attributes.

Reply:
I was also hoping to find a way that kept backwards compatibility. At least I'm hoping breaking this will make it a bit easier going forward.

Do you mean restrict known attributes like emitting `"has no attribute"` errors?

Then, calling wise, if we replace

```python
WithAnnotations[MyModel] -> MyModel@AnnotatedWith[Any]
```

We can require that `.annotate` has been called, but we don't care with what. Just like before. __I'm thinking this is the implementation we want__.

Else if we replace like (this is what we do now)

```python
WithAnnotations[MyModel] -> MyModel
```

This wouldn't distinguish between having called `.annotate` or not. Meaning that there's no requirement of having called `.annotate` but having done it is also accepted.